src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -649,10 +652,13 @@ ConfigSelector::CallConfig XdsResolver::XdsConfigSelector::GetCallConfig(
         case XdsApi::Route::HashPolicy::HEADER:
           new_hash = HeaderHashHelper(hash_policy, args.initial_metadata);
           break;
-        case XdsApi::Route::HashPolicy::CHANNEL_ID:
-          new_hash =
-              static_cast<uint64_t>(reinterpret_cast<uintptr_t>(resolver));
+        case XdsApi::Route::HashPolicy::CHANNEL_ID: {
+          std::string address_str = absl::StrFormat(
```

Comment:
Remind me why we're not just using the address directly here?  It's not clear to me what we gain by printing the address and sending it through the hash function, because that won't make the result any better distributed.

Reply:
because I found that the address is always too small of a number that it was only matching the lowest entry in the ring

Reply:
But this change just maps that to another fixed number, so it will still be just as deterministic; it will just always be a number in the middle of the ring instead of always being a number at the start of the ring.  I don't see how that helps.

What actually breaks if you just use the address directly?  Do tests fail?

If what we need here is more randomization, then maybe an alternative would be to just use a random number the same way that we're doing when there is no hash policy (i.e., calling `rand()` twice, once for high-order bits and once for low-order bits) and storing that value in the resolver so that we use the same value for all RPCs with this hash policy.

Reply:
hum.. you are right, i did not add more randomization.

ok will just use the address, tests are fine